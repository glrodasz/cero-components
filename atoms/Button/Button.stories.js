import { Button, options, styles } from '.'
import {
  getTemplate,
  getListTemplate,
  getOptionsArgTypes,
} from '../../helpers/storybook'

const Template = getTemplate(Button, styles)
const ListTemplate = getListTemplate(Button, styles)

export default {
  title: 'Atoms/Button',
  component: Button,
  args: {
    children: 'You Shall Not Pass!',
  },
  argTypes: {
    type: getOptionsArgTypes(options.types),
    children: { control: 'text' },
    isMuted: { description: 'It requires `type` to be "primary"' },
    onClick: { defaultValue: null },
  },
}

export const Default = Template.bind({})

export const Types = ListTemplate.bind({})
Types.args = { items: options.types.map((type) => ({ type })) }

export const Inline = Template.bind({})
Inline.args = { isInline: true }

export const Muted = Template.bind({})
Muted.args = { type: 'primary', isMuted: true }
