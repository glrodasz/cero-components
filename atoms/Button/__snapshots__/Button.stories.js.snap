// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/Button Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <button
    className="button type-primary"
    onClick={[Function]}
  >
    <header
      className="heading color-inverted size-md weight-bold"
    >
      You Shall Not Pass!
    </header>
  </button>
</div>
`;

exports[`Storyshots Atoms/Button Inline 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <button
    className="button type-primary is-inline"
    onClick={[Function]}
  >
    <header
      className="heading color-inverted size-md weight-bold"
    >
      You Shall Not Pass!
    </header>
  </button>
</div>
`;

exports[`Storyshots Atoms/Button Muted 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <button
    className="button type-primary is-muted"
    onClick={[Function]}
  >
    <header
      className="heading color-inverted size-md weight-bold"
    >
      You Shall Not Pass!
    </header>
  </button>
</div>
`;

exports[`Storyshots Atoms/Button Types 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <button
    className="button type-primary"
    onClick={[Function]}
  >
    <header
      className="heading color-inverted size-md weight-bold"
    >
      You Shall Not Pass!
    </header>
  </button>
  <button
    className="button type-secondary"
    onClick={[Function]}
  >
    <header
      className="heading color-primary size-md weight-bold"
    >
      You Shall Not Pass!
    </header>
  </button>
  <button
    className="button type-tertiary is-inline"
    onClick={[Function]}
  >
    <header
      className="heading color-primary size-md weight-bold"
    >
      You Shall Not Pass!
    </header>
  </button>
</div>
`;
